pandas/core/arrays/categorical.py

```diff
@@ -1343,16 +1353,13 @@ def searchsorted(self, value, side='left', sorter=None):
                              ".as_ordered() to change the Categorical to an "
                              "ordered one")
 
-        from pandas.core.series import Series
-
-        values_as_codes = _get_codes_for_values(Series(value).values,
-                                                self.categories)
-
-        if -1 in values_as_codes:
-            raise ValueError("Value(s) to be inserted must be in categories.")
+        if is_scalar(value):
+            codes = self.categories.get_loc(value)
+        else:
+            codes = [self.categories.get_loc(val) for val in value]
```

Comment:
you can use ``.get_indexer`` here

Reply:
Unfortunately ``get_indexer`` is much slower than ``get_loc``:

```python
>>> %timeit c.categories.get_loc('b')
6.12 µs  # this PR
>>> %timeit c.categories.get_indexer(['b'])
257 µs
```
 I've made the update to use ``.get_indexer`` anyway, and will use this as an opportunity to look for a way to make ``get_indexer`` faster, as that will yield benefits beyound ``.searchsorted``. Alternatively I can roll back this lat PR, and add the get_indexer part later, when I figure out why it's slow.

Reply:
.get_indexer is for many items when it will be *much* faster than an iteration of .get_loc, but for a small number of items the reverse maybe true, e.g. there will be a cross-over point.

Reply:
I don't think that is true here: ``get_loc`` makes a call to ``get_indexer``, so ``get_indexer``  shouldn't be slower, and the very least not this much slower. My guess is that there is some unneeded type conversion or parameter usage happening.

I'll look into to it. If everything is in get_indexer for the right reasons, I just won't pursue the case further.